### PR TITLE
Add support for advanced native when live-bidding

### PR DESCRIPTION
**Pull request by TurpIF**
JIRA: EE-1275

### CODE REVIEW

File: publisher-sdk/src/main/java/com/criteo/publisher/BidListener.kt [hunk 1]
```diff
@@ -16,14 +16,15 @@
...
+import androidx.annotation.NonNull
 import com.criteo.publisher.model.CdbResponseSlot
 
 /**
  * Callback through which bid responses will be served asynchronously.
  * Each of the callback methods will only be called once.
  */
 interface BidListener {
-  fun onBidResponse(cdbResponseSlot: CdbResponseSlot)
+  fun onBidResponse(@NonNull cdbResponseSlot: CdbResponseSlot)
```
Comment (Amokrane) [inline publisher-sdk/src/main/java/com/criteo/publisher/BidListener.kt:27]: What's the value of `@NonNull` annotation in Kotlin code?

**Review by Amokrane — CHANGES_REQUESTED** [review]

File: publisher-sdk/src/main/java/com/criteo/publisher/Standalone.kt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ */
+
+package com.criteo.publisher
+
+import com.criteo.publisher.annotation.OpenForTesting
+import com.criteo.publisher.model.AdUnit
+import com.criteo.publisher.model.Config
+
+@OpenForTesting
+internal class Standalone(
```
[the remark this answers is not in the document]
Comment (TurpIF) [reply publisher-sdk/src/main/java/com/criteo/publisher/Standalone.kt:24]: What about putting this logic in the `BidManager` ? You're right that that this API might be used by any integration but I think that `LiveBidder` is not the right name: it is not always doing live bids.
Moreover, if later we drop the old bidding way, then this method would just delegate to the `BidManager`.

File: publisher-sdk/src/main/java/com/criteo/publisher/BidListener.kt (hunk 1 above)
Comment (TurpIF) [reply publisher-sdk/src/main/java/com/criteo/publisher/BidListener.kt:27]: image

You're right that's a mistake. My implementation automatically added a `@NonNull` and I was confused why the interface didn't had one. And I didn't get that it was Kotlin code, not Java.

File: publisher-sdk/src/main/java/com/criteo/publisher/Standalone.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (TurpIF) [reply publisher-sdk/src/main/java/com/criteo/publisher/Standalone.kt]: Done

File: publisher-sdk/src/main/java/com/criteo/publisher/BidManager.java [hunk 3]
```diff
@@ -125,6 +151,7 @@
    * @return a valid bid that may be displayed or <code>null</code> that should be ignored
    */
   @Nullable
+  // @VisibleForTesting TODO Migrate callers on #getBidForAdUnit
```
Comment (Amokrane) [inline publisher-sdk/src/main/java/com/criteo/publisher/BidManager.java:154]: Can you elaborate on the TODO more? JIRA ticket?

**Review by Amokrane — CHANGES_REQUESTED** [review]

File: publisher-sdk/src/main/java/com/criteo/publisher/BidManager.java (hunk 3 above)
Comment (TurpIF) [reply publisher-sdk/src/main/java/com/criteo/publisher/BidManager.java]: Sure, updated

**Review by Amokrane — APPROVED** [review]

**Pull request merged**